server/core_benchmarks_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2023 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !skip_js_tests && !skip_js_cluster_tests && !skip_js_cluster_tests_2
+// +build !skip_js_tests,!skip_js_cluster_tests,!skip_js_cluster_tests_2
+
+package server
+
+import (
+	"crypto/rand"
+	"crypto/tls"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/nats-io/nats.go"
+)
+
+func BenchmarkTLSReqReply(b *testing.B) {
+	const (
+		subject    = "test-subject"
+		configPath = "./configs/tls"
+	)
+
+	// default TLS client connection options
+	defaultOpts := []nats.Option{
+		nats.Secure(
+			&tls.Config{
+				InsecureSkipVerify: true,
+			},
+		),
+	}
+
+	schemes := []string{
```

Comment:
Is there a precedent for calling these "schemes"? They seem like key types to me

Reply:
Yes perhaps scheme isn't the best name. What about `ciphers`?

Reply:
They are not ciphers